pkg/serverless/metrics/enhanced_metrics.go

```diff
@@ -31,6 +32,25 @@ func getOutOfMemorySubstrings() []string {
 	}
 }
 
+// GenerateRuntimeDurationMetric generates the runtime duration metric
+func GenerateRuntimeDurationMetric(start time.Time, end time.Time, status string, tags []string, metricsChan chan []metrics.MetricSample) {
+	// first check if both date are set
+	if start.IsZero() || end.IsZero() {
+		log.Debug("Impossible to compute aws.lambda.enhanced.runtime_duration due to an invalid interval")
+	} else {
+		duration := end.Sub(start).Milliseconds()
+		// TODO do we want to do something with status (success/error/timeout) here ?
```

Comment:
Eventually I think we will want to generate the error metric here, but for now I would remove the TODO comment just to keep the codebase cleaner